src/main/java/org/opensearch/ad/task/ADBatchTaskCache.java

```diff
@@ -95,23 +90,14 @@ protected ADBatchTaskCache(ADTask adTask) {
             .compact(true)
             .precision(Precision.FLOAT_32)
             .boundingBoxCacheFraction(AnomalyDetectorSettings.BATCH_BOUNDING_BOX_CACHE_RATIO)
-            // same with dimension for opportunistic memory saving
-            // Usually, we use it as shingleSize(dimension). When a new point comes in, we will
-            // look at the point store if there is any overlapping. Say the previously-stored
-            // vector is x1, x2, x3, x4, now we add x3, x4, x5, x6. RCF will recognize
-            // overlapping x3, x4, and only store x5, x6.
-            .shingleSize(shingleSize)
+            // for external shingling, rcf does not recognize shingle. Thus, shingle size
+            // is 1 here.
+            // shingle in detector config and shingle size here are different things.
+            // shingle size in detector config impacts dimensions.
+            .shingleSize(1)
+            .anomalyRate(1 - AnomalyDetectorSettings.THRESHOLD_MIN_PVALUE)
```

Comment:
anomaly rate should be close to zero. Where is AnomalyDetectorSettings.THRESHOLD_MIN_PVALUE?

Reply:
Resolved. AnomalyDetectorSettings.THRESHOLD_MIN_PVALUE = 0.995 from here
https://github.com/opensearch-project/anomaly-detection/blob/main/src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java#L250

Reply:
Yeah, I am reusing a setting from Kll model.